geowebcache/sqlite/src/main/java/org/geowebcache/sqlite/MbtilesBlobStore.java

```diff
@@ -0,0 +1,538 @@
+                    listeners.sendTileStored(tile);
+                } else {
+                    // this an update
+                    listeners.sendTileUpdated(tile, olData.length);
+                }
+            } catch (Exception exception) {
+                throw Utils.exception(exception, "Error saving tile '%s' in file '%s'.", tile, file);
+            }
+        });
+    }
+
+    @Override
+    public boolean get(TileObject tile) throws StorageException {
+        Utils.updateLogId("GET-" + tile.toString());
+        File file = fileManager.getFile(tile);
+        Utils.debug(LOGGER, "Tile mapped to file '%s'.", file);
+        initDatabaseFileIfNeeded(file, tile.getLayerName(), tile.getBlobFormat());
+        // do work in readonly mode
+        boolean exists = connectionManager.doWork(file, true, connection -> {
+            // instantiating geotools mbtiles reader
+            MBTilesFile mbtiles = GeoToolsMbtilesUtils.getMBTilesFile(connection, file);
+            try {
+                // loading the tile using geotools reader
+                MBTilesTile gtTile = mbtiles.loadTile(tile.getXYZ()[2], tile.getXYZ()[0], tile.getXYZ()[1]);
+                if (gtTile.getData() != null) {
+                    // the tile exists lets add the data to the tile object
+                    tile.setBlob(Utils.byteArrayToResource(gtTile.getData()));
+                    Utils.debug(LOGGER, "Tile found on file '%s'.", file);
+                    return true;
+                }
+            } catch (Exception exception) {
+                throw Utils.exception(exception, "Error loading tile '%s' from MBTiles file '%s'.", tile, file);
+            }
+            Utils.debug(LOGGER, "Tile not found on file '%s'.", file);
+            return false;
+        });
+        if (exists && useCreateTime) {
+            // the tile exists and we need to set the create time in the tile object
+            Long createdTime = getTileCreateTime(file, tile.getXYZ()[2], tile.getXYZ()[0], tile.getXYZ()[1]);
+            if (createdTime == null) {
+                // no create time associated with this tile lets assume the current as the create time
+                createdTime = System.currentTimeMillis();
+                // update the create time
+                putTileCreateTime(file, tile.getXYZ()[2], tile.getXYZ()[0], tile.getXYZ()[1], createdTime);
+            }
+            tile.setCreated(createdTime);
+        } else if (exists) {
+            // we don't care about the create time, tile will never expire
+            tile.setCreated(System.currentTimeMillis());
+        }
+        return exists;
+    }
+
+    @Override
+    public boolean delete(TileObject tile) throws StorageException {
+        Utils.updateLogId("DELETE-" + tile.toString());
+        File file = fileManager.getFile(tile);
+        Utils.debug(LOGGER, "Tile mapped to file '%s'.", file);
+        initDatabaseFileIfNeeded(file, tile.getLayerName(), tile.getBlobFormat());
```

Comment:
Hmm... if the tile is not there do we need to init it to remove a tile that won't be inside it? :-)